protocols/request-response/src/lib.rs

```diff
@@ -922,7 +933,8 @@ where
 const EMPTY_QUEUE_SHRINK_THRESHOLD: usize = 100;
 
 /// Internal information tracked for an established connection.
-struct Connection {
+#[derive(Clone)]
+pub struct Connection {
```

Comment:
This had to be `pub` for: https://github.com/libp2p/rust-libp2p/blob/9c75886d2e4bb2f496516c50d449548a5714814b/protocols/request-response/src/lib.rs#L355-L359